pymc_marketing/clv/models/beta_geo.py

```diff
@@ -174,35 +205,22 @@ def logp(t_x, x, a, b, r, alpha, T):
 
             pm.Potential(
                 "likelihood",
-                logp(x=frequency, t_x=recency, a=a, b=b, alpha=alpha, r=r, T=T),
+                logp(
+                    x=self.frequency,
+                    t_x=self.recency,
+                    a=a,
+                    b=b,
+                    alpha=alpha,
+                    r=r,
+                    T=self.T,
+                ),
             )
 
-    def _process_priors(self, a_prior, b_prior, alpha_prior, r_prior):
-        # hyper priors for the Gamma params
-        if a_prior is None:
-            a_prior = pm.HalfFlat.dist()
-        else:
-            self._check_prior_ndim(a_prior)
-        if b_prior is None:
-            b_prior = pm.HalfFlat.dist()
-        else:
-            self._check_prior_ndim(b_prior)
-
-        # hyper priors for the Beta params
-        if alpha_prior is None:
-            alpha_prior = pm.HalfFlat.dist()
-        else:
-            self._check_prior_ndim(alpha_prior)
-        if r_prior is None:
-            r_prior = pm.HalfFlat.dist()
-        else:
-            self._check_prior_ndim(r_prior)
-
-        return super()._process_priors(a_prior, b_prior, alpha_prior, r_prior)
-
     def _unload_params(self):
-        trace = self.fit_result.posterior
-
+        try:
```

Comment:
Why this try except?

Reply:
It's a weird thing with the MAP sampling and creating the inferencedata out of it, for reason I couldn't find it creates the posterior with one more nested 'posterior' inside. So depending on the selected sampling once the .posterior postfix was needed and once it wasn't. It's something that for sure should be fixed, but I wanted to get more reviews faster, and not delay it to fix this minor thing